modules/appeals_api/spec/services/appeals_api/pdf_construction/generator/generator_hlr_spec.rb

```diff
@@ -0,0 +1,167 @@
+          expect(generated_pdf).to match_pdf(expected_pdf)
+        end
+      end
+
+      context 'with extra content' do
+        let(:fixture_name) { 'expected_200996_extra.pdf' }
+        let(:hlr) { create("extra_higher_level_review_#{opts[:api_version]}".to_sym, created_at:) }
+
+        # FIXME: This is skipped when api_version is v0 because the v0 output contains inconsistent whitespace
+        # compared to the v2 output. There is no actual visual difference in outputs regardless of the api_version
+        # in use, but these tests will fail nonetheless if they run where api_version = v0.
+        it 'generates the expected pdf', skip: opts[:api_version] == 'v0' do
+          expect(generated_pdf).to match_pdf(expected_pdf)
+        end
+      end
+
+      context 'with max length' do
+        let(:fixture_name) { 'expected_200996_maxlength.pdf' }
+        let(:hlr) do
+          (opts[:form_data] || {}).each do |field_name, field_value|
+            allow_any_instance_of(
+              "AppealsApi::PdfConstruction::HigherLevelReview::#{opts[:pdf_version].upcase}::FormData".constantize
+            ).to receive(field_name).and_return(field_value)
+          end
+
+          create(:extra_higher_level_review_v2, created_at:) do |appeal|
+            appeal.form_data = override_max_lengths(
+              appeal,
+              read_schema('200996.json', opts[:api_name], opts[:api_version])
+            )
+            # TODO: update countryCodeISO2 in expected_200996_maxlength.pdf with expected override_max_lengths values
+            appeal.form_data['data']['attributes']['veteran']['address']['countryCodeISO2'] = 'US'
+            appeal.form_data['data']['attributes']['claimant']['address']['countryCodeISO2'] = 'US'
+            appeal.auth_headers.merge!(
+              {
+                'X-VA-First-Name' => 'W' * 30,
+                'X-VA-Middle-Initial' => 'W',
+                'X-VA-Last-Name' => 'W' * 40,
+                'X-VA-File-Number' => 'W' * 9,
+                'X-VA-SSN' => 'W' * 9,
+                'X-VA-Insurance-Policy-Number' => 'W' * 18,
+                'X-VA-NonVeteranClaimant-SSN' => 'W' * 9,
+                'X-VA-NonVeteranClaimant-First-Name' => 'W' * 255,
+                'X-VA-NonVeteranClaimant-Middle-Initial' => 'W',
+                'X-VA-NonVeteranClaimant-Last-Name' => 'W' * 255,
+                'X-Consumer-Username' => 'W' * 255,
+                'X-Consumer-ID' => 'W' * 255
+              }
+            )
+          end
+        end
+
+        # FIXME: This is skipped when api_version is v0 because the v0 output contains inconsistent whitespace
+        # compared to the v2 output. There is no actual visual difference in outputs regardless of the api_version
+        # in use, but these tests will fail nonetheless if they run where api_version = v0.
+        it 'generates the expected pdf', skip: opts[:api_version] == 'v0' do
+          expect(generated_pdf).to match_pdf(expected_pdf)
```

Comment:
In two cases (extra data and max content), I wasn't able to figure out how to make existing tests for HLR PDF generation pass when the HLRs are created with `api_version` = v0. In both cases:

- Actual PDF outputs for `api_version` = v0 and `api_version` = v2 were visually the same - couldn't figure out what was different
- When I inspected the diff'ed text content that rspec uses in `match_pdf` here, I found that it had somehow detected whitespace changes that did not appear visually

To me, this seems maybe similar to the caveat we already have here for local vs. CI test runs [here in the NOD PDF specs](https://github.com/department-of-veterans-affairs/vets-api/pull/13112/files#diff-883fabd212548c8ca843ec7b40d9fd2e81c44b2504b1c2bf4aeca77282978abdL101). I've left these tests skipped for now because I'm wondering if we can explore using some utility to diff the PDFs visually in tests (maybe we could convert the generated PDFs to images in specs and diff those? it would still require us to have some tool like imagemagick at minimum, which could make this difficult) instead of relying on the `match_pdf` matcher.

Reply:
My first thought would be to simply update the api-v0 gold master PDF and see if subsequent tests continue to balk at different whitespace changes. So long as the PDFs are visually what we expect, updating the fixture file is an easier option than getting into a rabbit hole of alternate pdf matching techniques (ask me how I know :P)

Reply:
Right now, there are no separate fixture files per-API. I could split them out, though - maybe I'll do that.

Reply:
I think that might be big set of changes, though - @cilestin what do you think of my addressing that as a follow-up?

Reply:
I think that's fine. We're caught in a spot where there are enough changes to our API-version data structures (e.g. country code ISO2 -> ISO3) that our generator tests can't ignore API-version any longer. Ideally once we've adhered to the LH API Standards, new API-versions won't affect the generator output any longer (i.e. so long as the PDF-version doesn't change, we should output the same file regardless of API-version), but I know that's not what happens in reality.

All this is to say that I think that we're at the point where having API- & PDF-coupled versions of our specs are necessary, so make that as easy on yourselves as possible :)